pkg/apis/monitoring/v1/types.go

```diff
@@ -1804,34 +1804,50 @@ type PrometheusRule struct {
 // +k8s:openapi-gen=true
 type PrometheusRuleSpec struct {
 	// Content of Prometheus rule file
+	// +listType=map
+	// +listMapKey=name
 	Groups []RuleGroup `json:"groups,omitempty"`
 }
 
 // RuleGroup and Rule are copied instead of vendored because the
 // upstream Prometheus struct definitions don't have json struct tags.
 
 // RuleGroup is a list of sequentially evaluated recording and alerting rules.
-// Note: PartialResponseStrategy is only used by ThanosRuler and will
-// be ignored by Prometheus instances.  Valid values for this field are 'warn'
-// or 'abort'.  More info: https://github.com/thanos-io/thanos/blob/main/docs/components/rule.md#partial-response
 // +k8s:openapi-gen=true
 type RuleGroup struct {
-	Name                    string `json:"name"`
-	Interval                string `json:"interval,omitempty"`
-	Rules                   []Rule `json:"rules"`
+	// Name of the rule group.
+	Name string `json:"name"`
```

Comment:
Prometheus/Thanos won't allow an empty group name so we need some validation here.

```suggestion
	// +kubebuilder:validation:MinLength=1
	Name string `json:"name"`
```